internal/daemon/daemon_test.go

```diff
@@ -725,29 +724,39 @@ func (s *daemonSuite) TestRestartSystemWiring(c *check.C) {
 }
 
 func (s *daemonSuite) TestRebootHelper(c *check.C) {
-	cmd := testutil.FakeCommand(c, "shutdown", "", true)
-	defer cmd.Restore()
-
-	tests := []struct {
-		delay    time.Duration
-		delayArg string
-	}{
-		{-1, "+0"},
-		{0, "+0"},
-		{time.Minute, "+1"},
-		{10 * time.Minute, "+10"},
-		{30 * time.Second, "+0"},
-	}
+	var lastDelay time.Duration
+	var didSleep, didReboot bool
 
-	for _, t := range tests {
-		err := reboot(t.delay)
-		c.Assert(err, check.IsNil)
-		c.Check(cmd.Calls(), check.DeepEquals, [][]string{
-			{"shutdown", "-r", t.delayArg, "reboot scheduled to update the system"},
-		})
+	oldTimeSleep := timeSleep
+	oldSyscallReboot := syscallReboot
+	defer func() {
+		timeSleep = oldTimeSleep
+		syscallReboot = oldSyscallReboot
+	}()
 
-		cmd.ForgetCalls()
+	timeSleep = func(d time.Duration) {
+		didSleep = true
```

Comment:
Nit, but it's probably slightly better to use `numSleeps` and `numReboots` counters instead of `didX` so we know these are only called once.